### DISCUSSION TITLE
Bounding Volume

### DISCUSSION THREAD

**Discussion by kbrandwijk** [StjerneIdioten/I3D-Blender-Addon#135] [Ideas & Feature Requests]
I ran into a situation where I require to manually set the bounding volume  properties on a shape. I have checked the sources and the property panels, but I can't find any reference to it. 

```xml
<IndexedTriangleSet name="cube" shapeId="1" meshUsage="0" bvCenter="1 2 3" bvRadius="2" isOptimized="true">
```

I'm referring to the bvCenter and bvRadius properties on a shape. It would be very useful if these could be set directly in Blender.

Main usecase: any shape where a shader affects it's position (such as with continuous tracks) needs manually adjusted bounding volume position and radius.

![image](https://user-images.githubusercontent.com/852069/121789593-7ff37180-cbd7-11eb-9c00-96650cfdd4ce.png)

For example, white bv orb is where GE _thinks_ the BV is, while in fact is has to be around the track shape on the right, like this:

![image](https://user-images.githubusercontent.com/852069/121789621-ae714c80-cbd7-11eb-89cc-66cf146576e8.png)

Setting this in Blender directly will help the workflow  a lot.

In Maya, I found out you can create a cube, and set in the properties that it defines the bounds for another object 

![image](https://user-images.githubusercontent.com/852069/121789920-761f3d80-cbda-11eb-8d61-7a90e93da57b.png)

That would be nice (it could set the display to wireframe when you select this in the options panel), but I'd take just manual input fields as well.

**Comment by StjerneIdioten** [comment]
You are correct that there is nothing in the source code about bounding volumes, because they are in fact not calculated at all by the addon 😉  GE does it automatically when you save the I3D into its binary representation, so it's one of those things that has been pushed back as there really wasn't a need (You would always have to save as binary through GE anyway)

It would be fairly easy to mimic something like the maya way, but I am thinking maybe the opposite way around? So you get a custom "Bounding Box" property on your object, where you then just drop in another object to use as that objects bounding box. Blender has pointer properties for objects that gives you the possibility of either dragging and dropping or selecting from a list of objects in the scene and it could also trigger an automatic switch to wireframe mode of the dropped in object 😄

**Reply by kbrandwijk** [reply]
I agree that would be better than doing it the other way around. Having the property on the object seems more natural (which is generally the opposite of Maya).

**Comment by ssnd292** [comment]
https://github.com/StjerneIdioten/I3D-Blender-Addon/pull/164

**Comment by ssnd292** [comment]
![2023-04-09 13_53_56-GIANTS Editor 64bit 9 0 3 - C__Users_ssnd2_Desktop_BoundingBox_BoundingBox_Expor](https://user-images.githubusercontent.com/3914410/230771189-de1eeedb-fe01-4def-b315-77075619b2e9.jpg)

**Comment by StjerneIdioten** [comment]
[This release](https://github.com/StjerneIdioten/I3D-Blender-Addon/releases/tag/v4.3.0) adds support for using other objects as a bounding volume reference
